lib/clients/AzureClient.py

```diff
@@ -0,0 +1,453 @@
+from azure.common.credentials import ServicePrincipalCredentials
+from azure.mgmt.compute import ComputeManagementClient
+from azure.mgmt.compute.models import StorageAccountTypes
+from azure.storage.blob import BlockBlobService
+from msrestazure.azure_exceptions import CloudError
+from azure.mgmt.compute.models import DiskCreateOption
+from azure.mgmt.compute.models import DiskCreateOptionTypes
+import time
+import glob
+from .BaseClient import BaseClient
+from ..models.Snapshot import Snapshot
+from ..models.Volume import Volume
+from ..models.Attachment import Attachment
+
+
+class AzureClient(BaseClient):
+    def __init__(self, operation_name, configuration, directory_persistent, directory_work_list, poll_delay_time,
+                 poll_maximum_time):
+        super(AzureClient, self).__init__(operation_name, configuration, directory_persistent, directory_work_list,
+                                          poll_delay_time, poll_maximum_time)
+        self.subscription_id = configuration['subscription_id']
+        self.__azureCredentials = ServicePrincipalCredentials(
+            client_id=configuration['client_id'],
+            secret=configuration['client_secret'],
+            tenant=configuration['tenant_id']
+        )
+        self.resource_group = configuration['resource_group']
+        self.storage_account_name = configuration['storageAccount']
+        self.storage_account_key = configuration['storageAccessKey']
+
+        self.block_blob_service = BlockBlobService(
+            account_name=self.storage_account_name, account_key=self.storage_account_key)
+        self.compute_client = ComputeManagementClient(
+            self.__azureCredentials, self.subscription_id)
+
+        # +-> Check whether the given container exists
+        container_details = self.get_container()
+        if not container_details:
+            msg = 'Could not find or access the given container.'
+            self.last_operation(msg, 'failed')
+            raise Exception(msg)
+
+    def get_container(self):
```

Comment:
I am wondering if the name is correct. I know it is inherited from previous code, but this method really does a lot more that only get_container, may be have get_container only to do `self.block_blob_service.get_container_properties(self.CONTAINER)` and then rename to something else.

Reply:
done